tket2-hseries/src/extension/result.rs

```diff
@@ -161,11 +161,27 @@ impl ResultOpDef {
     }
 
     fn result_signature(&self) -> SignatureFunc {
-        PolyFuncType::new(
-            [vec![TypeParam::String], self.type_params()].concat(),
-            Signature::new(self.arg_type(), type_row![]),
+        if self.is_array_result_op() {
+            // Arrays need to be returned to preserve no-implicit copy guarantees.
```

Comment:
There is not really a "no-implicit copy guarantee" in tket2 -- that's a guppy thing, right? I'm not sure how to phrase this comment in a way that makes sense for tket2. Maybe something along the lines of "Do not consume input arrays to allow linearity"?